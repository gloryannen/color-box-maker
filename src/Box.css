.boxMain {
  display: inline-block;
  position: relative;
}

.boxText {
  color: white;
  position: absolute;
  bottom: 0;
  margin-bottom: 5px;
  margin-left: 5px;
}

.boxRemoveBtn {
  position: absolute;
  top: 0;
  right: 0;
  cursor: pointer;
  border-left: 1px solid rgb(255, 255, 255);
  border-bottom: 1px solid rgb(255, 255, 255);
  font-size: 14px;
  margin-top: 5px;
  margin-right: 5px;
  padding: 0.3em;
  background-color: transparent;
  color: rgb(255, 255, 255);
}
