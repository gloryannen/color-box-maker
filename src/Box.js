import React from 'react';
import './Box.css'

// Box - define what values are needed to compose one
const Box = ({ backgroundColor, width, height, id, handleRemoveBox }) => {
  const removeBox = () => handleRemoveBox(id);

  // Renders a Box with values passed to style attributes
  return (
    <div className='boxMain'>
      <div style={ {
        backgroundColor,
        width: `${width}px`,
        height: `${height}px`
      } }
      >
        <p className='boxText'>{ width } x { height }</p>
        <button className='boxRemoveBtn' onClick={ removeBox }>X</button>
      </div>
    </div>
  )

}

export default Box;