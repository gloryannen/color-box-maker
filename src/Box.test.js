import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import Box from './Box.js';

// Smoke test
it("renders without crashing", function () {
  render(<Box />);
});

// Snapshot test
it("matches snapshot", function () {
  const { asFragment } = render(<Box />);
  expect(asFragment()).toMatchSnapshot();
});
