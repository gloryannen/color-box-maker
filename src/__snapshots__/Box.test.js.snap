// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="boxMain"
  >
    <div>
      <p
        class="boxText"
      >
         x 
      </p>
      <button
        class="boxRemoveBtn"
      >
        X
      </button>
    </div>
  </div>
</DocumentFragment>
`;
